src/keep-current-value.js

```diff
@@ -1,11 +1,21 @@
 var getViewValueFrom = {
   SELECT: function(element) {
     var options = element.find('option');
+    var values = [];
     for (var i = 0; i < options.length; i++) {
       var el = angular.element(options[i].outerHTML);
-      if ( el.attr('selected') ) return el.attr('value');
-    };
-    if ( options[0] ) return angular.element(options[0]).attr('value');
+      if (el.attr('selected')) {
+        values.push(el.attr('value'));
+      }
+    }
+    if (element.attr('multiple') === 'multiple') {
```

Comment:
I think you just need to check the presence the "multiple" attribute

`multiple="true"` and `multiple` should also work